### PR TITLE
fix(docs): fix dotnet missing links, remove auth token from docs

### PR DESCRIPTION
**Pull request by korniltsev**
- update docs on obtaining dotnet libraries
- remove auth token mentions from variaous places
- make tenant_id optinoal, comment it
- reformat rust docs

### CODE REVIEW

File: docs/sources/configure-client/language-sdks/dotnet.md [hunk 1]
```diff
@@ -11,13 +11,21 @@ aliases:
 
 ## How to add .NET profiling to your application
 
-1. Download `Pyroscope.Profiler.Native.so` and `Pyroscope.Linux.ApiWrapper.x64.so` from [latest release](https://github.com/pyroscope-io/pyroscope-dotnet/releases/)
+1. Obtain `Pyroscope.Profiler.Native.so` and `Pyroscope.Linux.ApiWrapper.x64.so` from the [latest tarbal](https://github.com/pyroscope-io/pyroscope-dotnet/releases/):
```
Comment (simonswine) [inline docs/sources/configure-client/language-sdks/dotnet.md:14]:
```suggestion
1. Obtain `Pyroscope.Profiler.Native.so` and `Pyroscope.Linux.ApiWrapper.x64.so` from the [latest tarball](https://github.com/pyroscope-io/pyroscope-dotnet/releases/):
```

**Review by simonswine — APPROVED** [review]
LGTM

File: docs/sources/configure-client/language-sdks/dotnet.md [hunk 2]
```diff
@@ -94,12 +102,10 @@ Pyroscope.Profiler.Instance.SetContentionTrackingEnabled(enabled);
 Pyroscope.Profiler.Instance.SetExceptionTrackingEnabled(enabled);
 ```
 
-It is possible to dynamically change auth tokens.
+It is possible to dynamically change Authorization credentials.
```
Comment (bryanhuhta) [inline docs/sources/configure-client/language-sdks/dotnet.md:105]:
```suggestion
It is possible to dynamically change authorization credentials.
```

File: docs/sources/configure-client/language-sdks/dotnet.md [hunk 3]
```diff
@@ -94,12 +102,10 @@ Pyroscope.Profiler.Instance.SetContentionTrackingEnabled(enabled);
...
 
 ```cs
-// Set Authorization Bearer token. Clear any previously set Authorization tokens.
-Pyroscope.Profiler.Instance.SetAuthToken(token);
-// Set Authorization Basic username and password. Clear any previously set Authorization tokens.
+// Set Authorization Basic username and password. Clear any previously set Authorization credentials.
```
Comment (bryanhuhta) [inline docs/sources/configure-client/language-sdks/dotnet.md:108]:
```suggestion
// Set Basic authorization username and password. Clears any previous authorization credentials.
```

**Review by bryanhuhta — APPROVED** [review]
Couple of minor nits, but looks good 👍

**Pull request merged**